tests/phpunit/includes/tests-pods-field-boolean.php

```diff
@@ -239,7 +239,139 @@ public function test_method_validate_exists() {
 	 */
 	public function test_method_validate() {
 
-		$this->assertTrue( $this->field->validate( 'foo' ) );
+		// All values are valid as they are parsed to integers (1 or 0).
+
+		$options = array(
+			'boolean_format_type' => 'radio',
+			'required'            => false,
+		);
+
+		// Empty values.
+		$this->assertTrue( $this->field->validate( true, null, $options ) );
+		$this->assertTrue( $this->field->validate( 1, null, $options ) );
+		$this->assertTrue( $this->field->validate( '1', null, $options ) );
+		$this->assertTrue( $this->field->validate( 'Yes', null, $options ) );
+		$this->assertTrue( $this->field->validate( 'On', null, $options ) );
+		$this->assertTrue( $this->field->validate( 'True', null, $options ) );
+
+		// Non empty values.
+		$this->assertTrue( $this->field->validate( false, null, $options ) );
+		$this->assertTrue( $this->field->validate( 0, null, $options ) );
+		$this->assertTrue( $this->field->validate( '0', null, $options ) );
+		$this->assertTrue( $this->field->validate( 'No', null, $options ) );
+		$this->assertTrue( $this->field->validate( 'Off', null, $options ) );
+		$this->assertTrue( $this->field->validate( 'False', null, $options ) );
+
+		// Other
+		$this->assertTrue( $this->field->validate( '', null, $options ) );
+		$this->assertTrue( $this->field->validate( 'Foobar', null, $options ) );
+
+		$options = array(
+			'boolean_format_type' => 'radio',
+			'required'            => true,
+		);
+
+		// Empty values.
+		$this->assertTrue( $this->field->validate( true, null, $options ) );
+		$this->assertTrue( $this->field->validate( 1, null, $options ) );
+		$this->assertTrue( $this->field->validate( '1', null, $options ) );
+		$this->assertTrue( $this->field->validate( 'Yes', null, $options ) );
+		$this->assertTrue( $this->field->validate( 'On', null, $options ) );
+		$this->assertTrue( $this->field->validate( 'True', null, $options ) );
+
+		// Non empty values.
+		$this->assertTrue( $this->field->validate( false, null, $options ) );
+		$this->assertTrue( $this->field->validate( 0, null, $options ) );
+		$this->assertTrue( $this->field->validate( '0', null, $options ) );
+		$this->assertTrue( $this->field->validate( 'No', null, $options ) );
+		$this->assertTrue( $this->field->validate( 'Off', null, $options ) );
+		$this->assertTrue( $this->field->validate( 'False', null, $options ) );
+
+		// Other
+		$this->assertTrue( $this->field->validate( '', null, $options ) );
```

Comment:
```suggestion
		$this->assertFalse( $this->field->validate( '', null, $options ) );
```

Reply:
The input is valid since it will parse to 0. A valid Boolean value.